xgi/core/dihypergraph.py

```diff
@@ -328,9 +322,8 @@ def add_node(self, node, **attr):
         If node is already in the dihypergraph, its attributes are still updated.
 
         """
-        if node not in self._node_in:
-            self._node_in[node] = set()
-            self._node_out[node] = set()
+        if node not in self._node:
```

Comment:
will this ever be true? since `self._node` should a 2-key dict, only `"in"` and `"out"` will be `in self._node`?

Reply:
It if it never true, the new "empty" 2-key dict will be create each time

Reply:
Oh wait, there's one 2-key dict per node? I thought it was `H._nodes["in"][node_id]` but it's `H._node[node_id]["in"]`.

Reply:
Correct. Yeah, I made the design decision to index by nodes first.